skimage/util/_coord_conv.py

```diff
@@ -0,0 +1,234 @@
+        image_out = np.swapaxes(image, 0, 1)
+    else:
+        image_out = None
+
+    return points_out, image_out
+
+
+def cart_to_rc(points=None, deltao=None, image=None):
+    """Convert image and/or points' coordinates from `x_c, y_c` to `r, c` convention.
+
+        .........       o--->....
+        ^ y_c   .       |   c   .
+        |       .  ==>  |       .
+        |   x_c .       v r     .
+        o--->....       .........
+
+    Parameters
+    ----------
+    points : (K, 2) ndarray, optional
+        Array of points' `x_c, y_c` Cartesian coordinates.
+    deltao : int, optional
+        Distance between `y_c` and `r` vectors' origins.
+        Optional for `points` convertion if `image` is specified.
+    image : (M, N[, C]) ndarray, optional
+        Image stored in `r, c` axes convention.
+
+    Returns
+    -------
+    points_out : (K, 2) ndarray
+        Array of `points` `r, c` coordinates.
+    image_out : (M, N[, C]) ndarray
+        Same data as `image`, but in `r, c` convention.
+    """
+    if points is not None:
+        if deltao is None:
+            if image is None:
+                msg = ('When converting coordinates, either `deltao` or `image` '
+                       'have to be specified.')
+                raise ValueError(msg)
+            else:
+                deltao = image.shape[1] - 1
+        r = deltao - points[:, 1]
+        c = points[:, 0]
+        points_out = np.stack((r, c)).T
+    else:
+        points_out = None
+
+    if image is not None:
+        image_out = np.swapaxes(np.flipud(image), 0, 1)
+    else:
+        image_out = None
+
+    return points_out, image_out
+
+
+def rc_to_cart(points=None, deltao=None, image=None):
+    """Convert image and/or points' coordinates from `r, c` to `x_c, y_c` convention.
+
+        o--->....       .........
+        |   c   .       ^ y_c   .
+        |       .  ==>  |       .
+        v r     .       |   x_c .
+        .........       o--->....
+
+    Parameters
+    ----------
+    points : (K, 2) ndarray, optional
+        Array of points' `r, c` coordinates.
+    deltao : int, optional
+        Distance between `r` and `y_c` vectors' origins.
+        Optional for `points` convertion if `image` is specified.
+    image : (M, N[, C]) ndarray, optional
+        Image stored in `x_c, y_c` axes convention.
+
+    Returns
+    -------
+    points_out : (K, 2) ndarray
+        Array of `points` `r, c` coordinates.
+    image_out : (M, N[, C]) ndarray
+        Same data as `image`, but in `r, c` convention.
+    """
+    if points is not None:
+        if deltao is None:
+            if image is None:
+                msg = ('When converting coordinates, either `deltao` or `image` '
+                       'have to be specified.')
+                raise ValueError(msg)
+            else:
+                deltao = image.shape[0] - 1
+        x_c = points[:, 1]
+        y_c = deltao - points[:, 0]
+        points_out = np.stack((x_c, y_c)).T
```

Comment:
This will definitely be more complicated to do in nd... LOL But still feasible I think. Also, I recently became aware of `np.column_stack`, which is probably what you want here.